From foo@baz Wed May  5 09:23:46 AM CEST 2021
From: Jianxiong Gao <jxgao@google.com>
Date: Thu, 29 Apr 2021 17:33:13 +0000
Subject: swiotlb: don't modify orig_addr in swiotlb_tbl_sync_single
To: stable@vger.kernel.org, hch@lst.de, marcorr@google.com, sashal@kernel.org
Cc: Jianxiong Gao <jxgao@google.com>, Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Message-ID: <20210429173315.1252465-8-jxgao@google.com>

From: Jianxiong Gao <jxgao@google.com>

commit: 16fc3cef33a04632ab6b31758abdd77563a20759

swiotlb_tbl_map_single currently nevers sets a tlb_addr that is not
aligned to the tlb bucket size.  But we're going to add such a case
soon, for which this adjustment would be bogus.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Acked-by: Jianxiong Gao <jxgao@google.com>
Tested-by: Jianxiong Gao <jxgao@google.com>
Signed-off-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Signed-off-by: Jianxiong Gao <jxgao@google.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 kernel/dma/swiotlb.c |    1 -
 1 file changed, 1 deletion(-)

--- a/kernel/dma/swiotlb.c
+++ b/kernel/dma/swiotlb.c
@@ -660,7 +660,6 @@ void swiotlb_tbl_sync_single(struct devi
 
 	if (orig_addr == INVALID_PHYS_ADDR)
 		return;
-	orig_addr += (unsigned long)tlb_addr & (IO_TLB_SIZE - 1);
 
 	switch (target) {
 	case SYNC_FOR_CPU:
